### PR TITLE
Alias to asymmetrically log and copy it to clipboard

### PR DESCRIPTION
**Pull request by lillialexis**
## Summary:
<Add your summary here>

Issue: XXX-XXXX

## Test plan:

### CODE REVIEW

File: .gitconfig.khan
Comment (lillialexis) [inline .gitconfig.khan:57]: I do wonder if there are any weird security issues with using the `tee /dev/tty` bit... 🤔

Comment (gjf20) [reply .gitconfig.khan:57]: I don't personally know!  Totally fine if you want to hold off on merging this to check in with the security team!

**Review by gjf20 — APPROVED** [review]
LGTM! But no rush in case you want to check in with others on the security concerns.

**Comment by lillialexis** [comment]
Google and chatgpt say it's fine... 🤔 

@csilvers can you think of any funniness that could happen by piping the output of `git log` through `tee /dev/tty`? I am overthinking this?

**Comment by gjf20** [comment]
@reidmit Hey Reid!  Just wanted to share this PR with you because I recall you mentioning something similar a few months back (though I can't find the slack thread).

File: .gitconfig.khan
Comment (lillialexis) [reply .gitconfig.khan:58]: Basically, I want the command to _both_ output to the terminal and copy it to the clipboard... when it was only sending it to the clipboard, it felt weird, because you also couldn't see what was getting sent.

The idea is that webapp folks could run this from the command-line on their deploy branches so they can easily copy the first line of their commit message so that they can paste it into Slack during a deploy without having stuff the extra tedious.

Comment (csilvers) [reply .gitconfig.khan:58]: > Basically, I want the command to both output to the terminal and copy it to the clipboard... when it was only sending it to the clipboard, it felt weird, because you also couldn't see what was getting sent.

Right, I got that.  But writing to /dev/tty is an unusual way to do that.  The right way to do it is to write to /dev/stdout.  But I don't know if /dev/stdout is an os x thing.  Though I think it is?

Since you are piping the output of stdout to somewhere else, though, you'll need to be careful about it.  I think you want to do something fancy, like this:
```
exec 3>&1; git log ... | tee /proc/self/fd/3 | pbpaste; exec 3>&-
```
What this does is copy stdout to fd 3 (so now fd 1 and fd3 both point to stdout).  Then you do git log and tee to fd 3, which means that tee is now writing to stdout twice, once on fd 1 and once on fd 3.  The one on fd1 goes to pbpaste due to the pipe.  The one on fd 3 stays going to stdout.  The final exec closes fd 3 (probably not necessary in this case, since I think this git command runs in a subshell.  But I'm not sure, so I'm doing it to be safe).

I think this can fail if fd 3 is already in use? Not an issue if you're in a subshell, but I don't know about otherwise.  But you can tree this and see how it works for you.

Comment (csilvers) [reply .gitconfig.khan:58]: (This assumes /proc/self/fd exists on os x.)

**Comment by reidmit** [comment]
Thanks @gjf20 - so exciting! Yeah, I have a little script that looks like this:

```
#!/bin/zsh

cd ~/khan/webapp

git log origin/master..origin/reid --no-merges --pretty='format:- %s' | pbcopy

echo "Copied to clipboard!"
```

File: .gitconfig.khan
Comment (lillialexis) [reply .gitconfig.khan:58]: @csilvers that's great feedback... I knew something felt off with just `dev/tty`. Seeing that @reidmit had a script, I just pivoted to this: https://github.com/Khan/our-lovely-cli/pull/839

**Pull request closed without merging**
